agent/async/lru_cache.go

```diff
@@ -71,15 +74,40 @@ func (lru *lruCache) Get(key string) (Value, bool) {
 	return entry.value, true
 }
 
+// Set sets the key-value pair in the cache
 func (lru *lruCache) Set(key string, value Value) {
 	lru.Lock()
 	defer lru.Unlock()
 
 	lru.cache[key] = &entry{value: value, added: time.Now()}
+
+	// Remove from the evict list if existed
+	for element := lru.evictList.Front(); element != nil; element = element.Next() {
```

Comment:
This is the same code you have in `Delete`.  Why not move it out to a separate function and then call it in both places?